model/src/main/kotlin/utils/OrtResultExtensions.kt

```diff
@@ -42,6 +43,18 @@ fun OrtResult.addPackageConfigurations(packageConfigurationProvider: PackageConf
     return copy(resolvedConfiguration = resolvedConfiguration.copy(packageConfigurations = packageConfigurations))
 }
 
+/**
```

Comment:
commit: One of the (my) goals when introducing IDs for providers was to allow for "updating" the curations for a specific provider ID. Looking at is function and the way it's being used in the following changes seems to move further away from that, as all existing curations get overwritten.

Have you considered this use case? / what's the plan for it?

Reply:
> Have you considered this use case? / what's the plan for it?

Yes, one of the goals of the refactoring is to implement this in a generic way. I'll contact you separately to make sure that it works for the orthw use case.